python/singa/layer.py

```diff
@@ -499,7 +581,7 @@ def initialize(self, x):
             self.kernel_size[0],
             self.kernel_size[1],
         )
-        w_name = self.name + Layer.sep + 'W'
+        w_name = self.get_param_name('W')
```

Comment:
we should avoid this. it means we may a parameter  self.W whose name is set to "Weights".  It is a bit confusing on the naming.

Reply:
Has addressed the problems mentioned above. Ready to merge.

Reply:
I mean if we use ` w_name = self.get_param_name('W')`, then we can also write ` w_name = self.get_param_name('Weights')`. 
I think we should not let the users name the param tensors to avoid such issue.

Reply:
So users name the param tensors, we should raise a warning?

Reply:
if a user set the name of a param explicitly, then we should not change it implicitly by prepending a prefix to it.
e.g., if a user call `W.name = "Weight"`, then we should not change it to "layer1.conv1.Weight"

Reply:
We can't implement like that. \_\_setattr\_\_ will be called if users want to set an attr. But if they modify the name of the Tensor like this`W.name = "Weight"`, \_\_setattr\_\_ won't be called. And we can't prevent users from changing the name by adding Tensor.\_\_setattr\_\_ because this tensor might not a param of a layer.

Reply:
That's just the current implementation.https://github.com/apache/singa/pull/731/files#diff-3240fd3b573c3eaf8f8d6da8470e68e7R232

Reply:
I missed your updates... 
I will merge it soon.

Reply:
Thanks😄